modules/internal/ChapelArray.chpl

```diff
@@ -1745,6 +1745,7 @@ module ChapelArray {
     }
 
     /* Return the number of times ``val`` occurs in the array. */
+    deprecated "'count' is deprecated"
```

Comment:
It might be good to add a hint about how to do this with a reduction. i.e: 'A.count(val)' is deprecated; please use '+ reduce (A == val)' instead"

Reply:
I've updated to:

"'count' is deprecated use a reduction like '+ reduce (A == val)' instead"

One issue with this is the error message won't be completely accurate as the user may have named their array something other than 'A'. I'm not aware of any way I can get the name of the actual array calling the method (and in some cases it might be a nameless temporary anyway).

Still I agree giving some kind of suggestion (even if it's not a perfect match) will be helpful.

Reply:
Yeah it would be cool if the error message included the actually variable name but I'm not sure how that would be done either. I think that message looks good though!